core/devdoc/on-out-process-gateway-modules.md

```diff
@@ -0,0 +1,484 @@
+
+This is an example of what the "loaders" entry will look like this:
+```JSON
+"loaders" : [
+    { 
+        "type" : "outprocess",
+        "name" : "outprocess"
+    }
+]
+```
+
+and an example of how the proxy module will be loaded :
+
+```JSON
+"modules" : [
+{ 
+    "name" : "outprocess_module",
+    "loader" : {
+        "name" : "outprocess"
+        "entrypoint" : {
+            "activation.type" : "none",
+            "control.id" : "outprocess_module_control",
+            "message.id" : "outprocess_module_message",
+            }
+        }
+    }
+    "args" : {...}
+}
+]
+```
+
+### Proxy Module Configuration
+
+The proxy module’s configuration will include the following information:
+
+-   **Control channel ID**
+
+    The proxy module will use a control channel to send control information to 
+    the module host process. This is a required argument. The identifier is any 
+    string that uniquely identifies the control channel. Duplicate strings will 
+    cause the module creation to fail.
+
+-    **Message ID**
+
+    The proxy module will send and receive gateway messages on sockets attached 
+    to this ID. This is optional in the module's JSON configuration. The 
+    identifier must be a unique string that identifies the message channel. 
+    Duplicate strings will cause the module creation to fail. If the 
+    ID is not set, the message channel URI will be composed from a UUID.
+
+-   **Activation Type**
+
+    This is an enumeration with values indicating how the hosting process will
+    be activated. It could indicate one of the following possible values:
+
+    -   **None:** An activation type of **None** indicates that the proxy module
+        does not have to do anything in particular to *activate* the module. The
+        expectation is that out-of-band measures will be adopted to ensure that
+        the out-proc module will be activated.
+
+    -   **Fork:** An activation type of **Fork** means that the proxy module
+        will attempt to fork the hosting process when the module is initialized.
+        The proxy will wait for a fixed duration in order to let the outprocess
+        process complete initialization and start listening on it’s `NN_RESP`
+        socket. Additional context information is supplied as part of the
+        configuration:
+
+        -   *Host Path*: Path to the executable that will host the module in
+            question
+        -   *Arguments*: array of command line arguments, if any.
+
+    -   **Service:** This activation type indicates that the module host is a
+        service and the proxy is expected to *start* the service during
+        initialization. The configuration might indicate whether this is an
+        *upstart* or *systemd* or a Windows system service.
+
+    -   **Container:** This activation type indicates that the out-proc module
+        is hosted in a Docker container and needs to be started during
+        initialization.
```

Comment:
Are these implemented already? If not I would put a tag saying (not yet implemented) or something. Just to set the right expectation....